server/src/main/java/org/elasticsearch/indices/IndicesService.java

```diff
@@ -1854,14 +1860,15 @@ public boolean allPendingDanglingIndicesWritten() {
     }
 
     /**
-     * @return the field type of the {@code @timestamp} field of the given index, or {@code null} if:
+     * @return a map holding the field types of the {@code @timestamp} and {@code event.ingested} fields of the given index,
      * - the index is not found,
      * - the field is not found, or
-     * - the field is not a timestamp field.
+     * - the mapping is not known yet, or
+     * - the index does not have a useful timestamp field.
      */
     @Nullable
-    public DateFieldMapper.DateFieldType getTimestampFieldType(Index index) {
-        return timestampFieldMapperService.getTimestampFieldType(index);
+    public Map<String, DateFieldMapper.DateFieldType> getTimestampFieldTypeMap(Index index) {
```

Comment:
Could we avoid having a map, especially as we know we support having only max two entries? I think that this signature is more generic than it needs to be, and it requires additional knowledge / context when looking at the code.